ledger/test-common/src/main/daml/PingPong.daml

```diff
@@ -0,0 +1,186 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+daml 1.2
+module PingPong where
+
+import DA.Text
+
+template Ping
+  with
+    id : Text
+    initiator : Party
+    responders : [Party]
+    maxLevel : Int
+  where
+
+    signatory initiator
+    observer responders
+
+    choice Respond : Either (ContractId Pong) (ContractId Explode)
+        with
+        winner : Party
+      controller winner
+        do
+        if maxLevel == 0 then do
+            png <- create Pong with id, initiator, responder = winner, observers = responders
+            return (Left png)
+        else do
+            expl <- create Explode with id, initiator, responders, path = "", maxLevel
+            return (Right expl)
```

Comment:
Can we remove the explosion logic until we need it? (And even if we do need it eventually, I'd rather make it a separate DAML model.)

Reply:
I'll have a look to understand it better but so far my understanding it that it builds a tree of pings rather than a single one. @mziolekda @dajmaki what do you think, are we going to use it soon e.g. to get a bit closer to real-world workload? If so I'd rather clean it up and leave it there.